runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -726,9 +733,17 @@ TR_J9ServerVM::canAllocateInlineClass(TR_OpaqueClassBlock *clazz)
    auto it = _compInfoPT->getClientData()->getROMClassMap().find((J9Class*) clazz);
    if (it != _compInfoPT->getClientData()->getROMClassMap().end())
       {
-      return it->second.inlineClass;
+      if (it->second.classInitialized)
+         {
+         return (it->second.romClass->modifiers & (J9_JAVA_ABSTRACT | J9_JAVA_INTERFACE ));
+         }
+      else
+         {
+         JITaaS::J9ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
```

Comment:
Here we also need to return the correct answer to the caller. If the class is initialized at this point we check againthe romClass->modifiers, otherwise we return false